src/entities.c

```diff
@@ -0,0 +1,2068 @@
+	"supsetneqq;",
+	"supsim;",
+	"supsub;",
+	"supsup;",
+	"swArr;",
+	"swarhk;",
+	"swarr;",
+	"swarrow;",
+	"swnwar;",
+	"szlig;",
+	"target;",
+	"tau;",
+	"tbrk;",
+	"tcaron;",
+	"tcedil;",
+	"tcy;",
+	"tdot;",
+	"telrec;",
+	"tfr;",
+	"there4;",
+	"therefore;",
+	"theta;",
+	"thetasym;",
+	"thetav;",
+	"thickapprox;",
+	"thicksim;",
+	"thinsp;",
+	"thkap;",
+	"thksim;",
+	"thorn;",
+	"tilde;",
+	"times;",
+	"timesb;",
+	"timesbar;",
+	"timesd;",
+	"tint;",
+	"toea;",
+	"top;",
+	"topbot;",
+	"topcir;",
+	"topf;",
+	"topfork;",
+	"tosa;",
+	"tprime;",
+	"trade;",
+	"triangle;",
+	"triangledown;",
+	"triangleleft;",
+	"trianglelefteq;",
+	"triangleq;",
+	"triangleright;",
+	"trianglerighteq;",
+	"tridot;",
+	"trie;",
+	"triminus;",
+	"triplus;",
+	"trisb;",
+	"tritime;",
+	"trpezium;",
+	"tscr;",
+	"tscy;",
+	"tshcy;",
+	"tstrok;",
+	"twixt;",
+	"twoheadleftarrow;",
+	"twoheadrightarrow;",
+	"uArr;",
+	"uHar;",
+	"uacute;",
+	"uarr;",
+	"ubrcy;",
+	"ubreve;",
+	"ucirc;",
+	"ucy;",
+	"udarr;",
+	"udblac;",
+	"udhar;",
+	"ufisht;",
+	"ufr;",
+	"ugrave;",
+	"uharl;",
+	"uharr;",
+	"uhblk;",
+	"ulcorn;",
+	"ulcorner;",
+	"ulcrop;",
+	"ultri;",
+	"umacr;",
+	"uml;",
+	"uogon;",
+	"uopf;",
+	"uparrow;",
+	"updownarrow;",
+	"upharpoonleft;",
+	"upharpoonright;",
+	"uplus;",
+	"upsi;",
+	"upsih;",
+	"upsilon;",
+	"upuparrows;",
+	"urcorn;",
+	"urcorner;",
+	"urcrop;",
+	"uring;",
+	"urtri;",
+	"uscr;",
+	"utdot;",
+	"utilde;",
+	"utri;",
+	"utrif;",
+	"uuarr;",
+	"uuml;",
+	"uwangle;",
+	"vArr;",
+	"vBar;",
+	"vBarv;",
+	"vDash;",
+	"vangrt;",
+	"varepsilon;",
+	"varkappa;",
+	"varnothing;",
+	"varphi;",
+	"varpi;",
+	"varpropto;",
+	"varr;",
+	"varrho;",
+	"varsigma;",
+	"vartheta;",
+	"vartriangleleft;",
+	"vartriangleright;",
+	"vcy;",
+	"vdash;",
+	"vee;",
+	"veebar;",
+	"veeeq;",
+	"vellip;",
+	"verbar;",
+	"vert;",
+	"vfr;",
+	"vltri;",
+	"vopf;",
+	"vprop;",
+	"vrtri;",
+	"vscr;",
+	"vzigzag;",
+	"wcirc;",
+	"wedbar;",
+	"wedge;",
+	"wedgeq;",
+	"weierp;",
+	"wfr;",
+	"wopf;",
+	"wp;",
+	"wr;",
+	"wreath;",
+	"wscr;",
+	"xcap;",
+	"xcirc;",
+	"xcup;",
+	"xdtri;",
+	"xfr;",
+	"xhArr;",
+	"xharr;",
+	"xi;",
+	"xlArr;",
+	"xlarr;",
+	"xmap;",
+	"xnis;",
+	"xodot;",
+	"xopf;",
+	"xoplus;",
+	"xotime;",
+	"xrArr;",
+	"xrarr;",
+	"xscr;",
+	"xsqcup;",
+	"xuplus;",
+	"xutri;",
+	"xvee;",
+	"xwedge;",
+	"yacute;",
+	"yacy;",
+	"ycirc;",
+	"ycy;",
+	"yen;",
+	"yfr;",
+	"yicy;",
+	"yopf;",
+	"yscr;",
+	"yucy;",
+	"yuml;",
+	"zacute;",
+	"zcaron;",
+	"zcy;",
+	"zdot;",
+	"zeetrf;",
+	"zeta;",
+	"zfr;",
+	"zhcy;",
+	"zigrarr;",
+	"zopf;",
+	"zscr;",
+	"zwj;",
+	"zwnj;",
+};
+
+typedef struct pascal_str {
+	const char* str;
+	size_t len;
+} pascal_str;
+
+static int
+bsearch_pascal_str_comp(const void *s1, const void *s2) {
+	const struct buf *entity_buf = s1;
+	const char * const *arg = s2;
+	return strncmp((const char*)entity_buf->data, *arg, entity_buf->size);
+}
+
+int
+is_allowed_named_entity(const struct buf *entity_buf) {
+	const void *found = bsearch(
```

Comment:
Nice! [According to the docs it can work with non-null terminated strings](https://www.gnu.org/software/gperf/manual/gperf.html#Binary-Strings), so we should still be able to avoid doing a copy, as well :smile: .